### PR TITLE
Json parsing

### PR DESCRIPTION
**Pull request by brentschmaltz**

### CODE REVIEW

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs [hunk 1]
```diff
@@ -40,57 +40,33 @@ namespace Microsoft.IdentityModel.Protocols.OpenIdConnect
...
+            return new OpenIdConnectConfiguration(json);
         }
 
         /// <summary>
         /// Serializes the <see cref="OpenIdConnectConfiguration"/> object to a json string.
         /// </summary>
         /// <param name="configuration"><see cref="OpenIdConnectConfiguration"/> object to serialize.</param>
         /// <returns>json string representing the configuration object.</returns>
+        /// <exception cref="ArgumentNullException">if 'configuration' is null or empty.</exception>
+        /// <exception cref="ArgumentException">if 'json' fails to deserialize.</exception>
```
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs]: The comment does not apply to this method.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs [hunk 2]
```diff
@@ -107,72 +83,35 @@ public OpenIdConnectConfiguration()
         /// Initializes an new instance of <see cref="OpenIdConnectConfiguration"/> from a json string.
         /// </summary>
         /// <param name="json">a json string containing the metadata</param>
-        /// <exception cref="ArgumentNullException">if 'json' is null or whitespace.</exception>
+        /// <exception cref="ArgumentNullException">if 'json' is null or empty.</exception>
         public OpenIdConnectConfiguration(string json)
+            : this()
```
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs]: Why chaining the constructor?

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs [hunk 3]
```diff
@@ -155,21 +138,7 @@ private void Copy(JsonWebKey key)
...
-            set
-            {
-                if (value == null)
-                    throw LogHelper.LogException<ArgumentNullException>(LogMessages.IDX10001, "KeyOps");
-
-                foreach (string keyOp in value)
-                    _keyops.Add(keyOp);
-            }
-        }
+        public IList<string> KeyOps { get; private set; } = new List<string>();
```
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: There is a change of behavior here, just want to make sure its intentional. Earlier, setting KeyOps would add values to existing array (always additive), now it replaces the whole list. The new behavior seems right to me, always additive does not make sense.

Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: Yep, glad you noticed. This is the correct behavior always return a valid collection and have users modify it.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs (hunk 2 above)
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs]: Will fix.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs (hunk 1 above)
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs]: I will remove this.

**Comment by tushargupta51** [comment]
:shipit:

**Comment by brentschmaltz** [comment]
also fixed
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/378

**Pull request merged**